### ISSUE TITLE
Font tags are not propergated for text in tables.

### ISSUE THREAD

**Issue by dmail00** [py-pdf/fpdf2#1013]
Font tags are correctly setting the font attributes[1], however these attributes are not transferred to the style. Below is a silly example which shows the issue, it sets the background and the font color to the same and therefore the font should not be seen, however the font color stays black, In reality you would not do this, but when changing the background color it will often happen that font color is changed for a better constrast.

[1] https://github.com/py-pdf/fpdf2/blob/master/fpdf/html.py#L523

**Minimal code**
Please include some minimal Python code reproducing your issue:
```python


def main() -> None:
    data = """<table>
    <thead>
        <tr bgcolor="#711C45">
            <th width="25%"><font color="#711C45">Mark</font></th>
        </tr>
    </thead>
    <tbody>
        <tr bgcolor="#E4E4E4"><td align="center"><font color="#E4E4E4">less than 2</font></td></tr>
    </tbody>
</table>"""

    test = FPDF()
    test.add_page()
    test.set_font("Helvetica", "")
    test.set_font_size(8)

    test.write_html(data)

    test.output("test.pdf")


if __name__ == "__main__":
    main()
...
```
If you don't know how to build a minimal reproducible example, please check this tutorial: https://stackoverflow.com/help/minimal-reproducible-example

**Environment**
Please provide the following information:
* **Operating System**: Windows
* **Python version**:  3.11.3
* **`fpdf2` version used**:  github head

**Comment by gmischler** [comment]
Thanks for reporting this, @dmail00 !

Unfortunately, the code generating the tables is not in an optimal state at the moment. Too many (sometimes conflicting) features have been added to it with not enough attention to how they all play together, and syntactic sugar has sometimes been given priority over a well structured API.

I am currently refactoring the tables to convert its cells to [text regions](https://py-pdf.github.io/fpdf2/TextRegion.html), which will make them a lot more flexible to use. Since this is only reasonably possible with a thorough code cleanup, it will likely result in some backwards incompatible changes, but that should be worth it if the result is a more consistently usable functionality.
All of that may take some time, but afterwards it will be much easier to fix this bug. It will also make it possible to use formatting tags like `<b>` within table cells, which is currently prohibited.

In short: We'll get there, but you'll need a little patience for this one...

**Comment by dmail00** [comment]
I pulled that down and it worked great. From just looking at the change I did not understand how until I debugged it. I can see that without a bgcolor or emphasis Row.cell correctly captured the required style :)
Thanks.

**Comment by Lucas-C** [comment]
Thank you for you feedback @dmail00

I will now merge PR #1015 and close this issue
